config/config.go

```diff
@@ -102,6 +112,9 @@ func (c *Config) Ensure(fromCloud bool, logger golog.Logger) error {
 			}
 			logger.Errorw("module config error; starting robot without module", "name", c.Modules[idx].Name, "error", err)
 		}
+		if err := c.validateUniqueResource(seenResources, c.Modules[idx].Name); err != nil {
+			logger.Error(err.Error())
```

Comment:
this was something i wasn't super sure about. Should we still just log this but not exit out of the validator here? It made sense to just keep current implementation for now, but at least warn the user that this happening in logs. Issues with overwrites however will still happen if there are resources with the same name.

Reply:
I'm not particularly concerned with breaking existing configs that have duplicate names. I think we should just return the error here instead of logging. @cheukt do you have any thoughts? If we have the option to stop construction (we usually don't from within the resource manager), I think we should here.

Reply:
Fragments create a non-insignificant risk of name collision. I was talking this over with @nataliajacobowitz who indicated that it is better to log an error and continue building. However, I can see the merits for erroring early rather than continuing with missing components... Might not be a bad idea to double check with fahmina or natalia.

Reply:
I think erroring out here as part of disabling partial start would be fine - it'd warn users that something is wrong